src/main/java/com/clouway/push/client/ChannelApiPushEventBus.java

```diff
@@ -40,20 +40,15 @@ public HandlerRegistration addHandler(final PushEvent.Type type, final PushEvent
 
     final HandlerRegistration[] handlerRegistration = {null};
 
-    if (!pushChannelApi.hasOpenedChannel()) {
+    if (!pushChannelApi.hasInitialConnection()) {
 
-      pushChannelApi.connect(new AsyncConnectCallback() {
+      pushChannelApi.connect();
```

Comment:
In class ChannelImpl are handled only events which are need at current time.

I think if opening of connection is failed opening of connection should retry in some time interval